pull_request_template.md

```diff
@@ -22,6 +22,18 @@ _https://github.com/flyteorg/flyte/issues/<number>_
 [ ] - Not correct; marked as **not** done
 -->
 
+### Why?
+
+<!-- Tell reviewers why you add this PR. -->
+
+### What?
+
+<!-- Tell reviewers what changes did you do -->
+
+### Tests
```

Comment:
We might not need `note to reviewers`, which should already be stated in `what` and `why` section

Reply:
NO PROBLEM!